packages/lisk-transactions/src/transactions/base.ts

```diff
@@ -57,6 +59,18 @@ export enum MultisignatureStatus {
 	UNKNOWN = 2,
 }
 
+export interface EntityMap {
+	readonly [name: string]: ReadonlyArray<unknown> | undefined;
+}
+
+export interface RequiredState {
```

Comment:
I think rather than exporting this interface, the interface for RequiredState in BaseTransaction class should only include properties required in the BaseTransaction class, whereas every child class should implement its own Interface dependent on the properties required for that transaction type. This will allow classes to have a strict type for parameters passed to the `verify`, `apply` and `undo` functions.
Therefore, in the BaseTransactions class, the RequiredState interface should only include `sender` object because the BaseTransaction class only makes use of `sender` object. In the child classes which require other objects like `recipient` and `dependentState`, they should define their own interface strictly matching the properties.
We will always be able to call `super` function from child classes because the RequiredState interface in the BaseTransaction class will always be less strict.

Reply:
I thought it would be hard to override the interface, but it seems like it works like a charm, so it's fixed to have only required values!